user_tools/src/spark_rapids_pytools/rapids/qualification.py

```diff
@@ -209,20 +209,37 @@ def _process_cpu_cluster_args(self, offline_cluster_opts: dict = None):
 
     def _process_gpu_cluster_args(self, offline_cluster_opts: dict = None) -> bool:
         gpu_cluster_arg = offline_cluster_opts.get('gpuCluster')
-        if gpu_cluster_arg is None:
-            self.logger.info('Creating GPU cluster by converting the CPU cluster instances to GPU supported types')
-            # Convert the CPU instances to support gpu
-            orig_cluster = self.ctxt.get_ctxt('cpuClusterProxy')
-            gpu_cluster_obj = self.ctxt.platform.migrate_cluster_to_gpu(orig_cluster)
-        else:
+        if gpu_cluster_arg:
             gpu_cluster_obj = self._create_migration_cluster('GPU', gpu_cluster_arg)
+        else:
+            orig_cluster = self.ctxt.get_ctxt('cpuClusterProxy')
+            gpu_cluster_obj = None
+            if orig_cluster:
+                # Convert the CPU instances to support gpu. Otherwise, gpuCluster is not set
+                self.logger.info('Creating GPU cluster by converting the CPU cluster instances to GPU supported types')
+                gpu_cluster_obj = self.ctxt.platform.migrate_cluster_to_gpu(orig_cluster)
         self.ctxt.set_ctxt('gpuClusterProxy', gpu_cluster_obj)
-        return True
+        return gpu_cluster_obj is not None
```

Comment:
If a user pass in `--gpu_cluster` but no `--cpu_cluster` by accident, is the qualification tool going to throw an error?

Reply:
Good question!
yes, this will cause a crash because we have a null object which is the cpu_cluster.
We avoid the crash by disabling savings calculations is either of CPU/GPU clusters are missing.

Reply:
Added a fix in the latest commit
Thanks @cindyyuanjiang